src/Maestro/maestro-angular/src/app/page/build-graph-table/build-graph-table.component.ts

```diff
@@ -102,6 +104,11 @@ function sortBuilds(graph: BuildGraph): BuildData[] {
   for (const node of result) {
     node.hasIncoherentDependencies = hasIncoherentDependencies(node.build, result, false);
     node.hasIncoherentDependenciesIncludingToolsets = hasIncoherentDependencies(node.build, result, true);
+
+    // If node is incoherent with product check to make sure there aren't any cycles in the dependencies
+    let cycles = hasCycles(node.build, result);
```

Comment:
This can be a tuple destructuring assignment: `let [hasCycle, cyclePath = <fn>`

Reply:
I was hoping there would be something like that.